### PR TITLE
Roots of Quadratic equation in java

### PR DESCRIPTION
**Pull request by nagasaisriya**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?
YES
(Write your answer here.)

### Description
Quadratic equations are actually used in everyday life, as when calculating areas, determining a product's profit or formulating the speed of an object.
Solving Quadratic equations in Java-A program that prints all real solutions to quadratic equation ax^2+bx+c=0, if discriminant is negative displays a message "roots are imaginary".
(Write your answer here.)

### Checklist

- [ X ] I've read the contribution guidelines.
- [ X ] I've checked the issue list before deciding what to submit.
- [ X ] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests
#2628 
(Write your answer here.)

### CODE REVIEW

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Few changes: 
- Add a new line at the EOF
- Use meaningful variable names
- Move your Code logic to separate function and use `main()` for the driver code

**Comment by nagasaisriya** [comment]
Yeah, would look into it. 
Thank you

**Comment by nagasaisriya** [comment]
I have updated the File according to requested changes, @HarshCasper do review it

File: Java/math/Quadratic_Equation.java [hunk 1]
```diff
@@ -0,0 +1,113 @@
+
```
Comment (ankitaggarwal23) [inline Java/math/Quadratic_Equation.java]: Remove that extra space

File: Java/math/Quadratic_Equation.java [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    a = sc.nextDouble();
+    System.out.println();
+    System.out.print("Enter b: ");
+    b = sc.nextDouble();
+    System.out.println();
+    System.out.print("Enter c: ");
+    c = sc.nextDouble();
+    System.out.println();
+    // If a is 0, then equation is not quadratic, but
+    // linear
```
Comment (ankitaggarwal23) [inline Java/math/Quadratic_Equation.java]: remove that and shift to upper comment

File: Java/math/Quadratic_Equation.java [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+    if (a == 0) {
+      System.out.println("Its a linear equation,Not a quadratic equation!");
+      return;
+    }
+    // calculate the discriminant (b^2 - 4ac)
```
Comment (ankitaggarwal23) [inline Java/math/Quadratic_Equation.java:30]: Remove redundant comments

File: Java/math/Quadratic_Equation.java [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+    double discriminant = b * b - 4 * a * c;
+
+    // check if discriminant is greater than 0
+    if (discriminant > 0) {
+
+      // two real and distinct roots
+      root1 = (-b + Math.sqrt(discriminant)) / (2 * a);
+      root2 = (-b - Math.sqrt(discriminant)) / (2 * a);
+      System.out.println("Roots are real and distinct");
+      System.out.format("root1 = %.2f and root2 = %.2f", root1, root2);
```
Comment (ankitaggarwal23) [inline Java/math/Quadratic_Equation.java]: Give one tab space inside the body of statement

File: Java/math/Quadratic_Equation.java [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+    }
+
+    // check if determinant is equal to 0
+    else if (discriminant == 0) {
+
+      // two real and equal roots
+      // discriminant is equal to 0
```
Comment (ankitaggarwal23) [inline Java/math/Quadratic_Equation.java]: Remove redundant comments as you already mentioned this line above

File: Java/math/Quadratic_Equation.java [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+
+    // if discriminant is less than zero
+    else {
+
+      // roots are complex number
+      double real = -b / (2 * a);
+      double imaginary = Math.sqrt(-discriminant) / (2 * a);
+      System.out.println("Imaginary roots!");
+      System.out.format("root1 = %.2f+%.2fi", real, imaginary);
+      System.out.format("\nroot2 = %.2f-%.2fi", real, imaginary);
```
Comment (ankitaggarwal23) [inline Java/math/Quadratic_Equation.java]: Give one tab space inside the body of statement

File: Java/math/Quadratic_Equation.java [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+    }
+
```
Comment (ankitaggarwal23) [inline Java/math/Quadratic_Equation.java]: Remove extra line

File: Java/math/Quadratic_Equation.java [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+  }
+
+  public static void main(String[] args) {
+    quadraticroots();
+
+  }
+}
```
Comment (ankitaggarwal23) [inline Java/math/Quadratic_Equation.java]: Give one tab space inside the body of statement and remove extra line

**Review by ankitaggarwal23 — COMMENTED** [review]
- Add Time and Space Complexity at the end of the Code   
- Use proper code Format 
- Remove extra lines of code and spaces

**Comment by nagasaisriya** [comment]
Hello, @ankitaggarwal23 @HarshCasper , made a commit and the updated file with requested changes, do review it. Thank you!

**Comment by HarshCasper** [comment]
> Hello, @ankitaggarwal23 @HarshCasper , made a commit and the updated file with requested changes, do review it. Thank you!

Not all of the comments have been addressed

**Comment by nagasaisriya** [comment]
> > Hello, @ankitaggarwal23 @HarshCasper , made a commit and the updated file with requested changes, do review it. Thank you!
> 
> Not all of the comments have been addressed

Please can you be more specific, like which comment is being missed and any other changes to be made?

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**